gulpfile.js

```diff
@@ -73,8 +79,12 @@ gulp.task('lint', function() {
 	];
 
 	return gulp.src(files)
-		.pipe(eslint())
+		.pipe(eslint({fix: argv.fix}))
```

Comment:
It makes it easier to fix linting errors. By running `gulp lint --fix`, the correction is applied automatically for all suitable linting rules (whitespaces, indentation, etc...).

Reply:
I get your point but this shouldn't be part of this PR which is already quite complex. Please revert.

Reply:
It's arguably more work than to just leave it here, but you are the boss :)
Should I also revert adding linting to typescript files then?

Reply:
I really want to keep this PR focused on migrating to v3 with minimum change as possible. I would actually have more feedback about this ESLint `--fix` but this would make this PR more difficult to understand and iterate. I will be happy to review another PR on master that adds support for `--fix`. However, I'm also thinking to get rid of gulp at some point so not sure it worth it.

> Should I also revert adding linting to typescript files then?

Yes please, for the same reasons but also because it's not consistent with the JS code and the previous TS formatting. I will introduce ESLint for TS in a separate PR.